src/iface/ethernet.rs

```diff
@@ -608,10 +616,26 @@ impl<'b, 'c> InterfaceInner<'b, 'c> {
             _ if handled_by_raw_socket =>
                 Ok(Packet::None),
 
-            // TODO: send error responses when appropriate.
-            _ => Ok(Packet::None)
+            _ => {
+                // Send back as much of the original payload as we can.
+                //
+                // ICMPv6 Contains no protocol unreachable reason for the
+                // destination unreachable message. Instead a parameter
+                // problem message is returned with the code set to
+                // "unrecognized next header" and the pointer set to the
+                // offset within the invoking packet.
+                let payload_len = cmp::min(ip_payload.len(),
+                                           IPV6_MIN_MTU - ipv6_repr.buffer_len() * 2 - 8);
```

Comment:
This formula is repeated four times (across IPv4 and IPv6). Should it be a function so it is not repeated?

Reply:
probably